### PR TITLE
Use the order of values in JWT_TOKEN_LOCATION to set order of precedence

### PR DESCRIPTION
**Pull request by stephendwolff**
I've added the update to run the token validation in the order set in JWT_TOKEN_LOCATION, and this allows a valid token earlier in the list to authenticate the user, and invalid to be ignored if later.

This suits my use case - but perhaps another option could be to require no invalid tokens?

I wrote a couple of tests, and fixed one of the error messages in the existing multiple token location test.

Also, I added a note to the configuration options documentation for JWT_TOKEN_LOCATION

### CODE REVIEW

File: flask_jwt_extended/view_decorators.py [hunk 1]
```diff
@@ -247,14 +247,21 @@ def _decode_jwt_from_json(request_type):
...
-    if config.jwt_in_headers:
-        get_encoded_token_functions.append(_decode_jwt_from_headers)
-    if config.jwt_in_json:
-        get_encoded_token_functions.append(lambda: _decode_jwt_from_json(request_type))
+
+    locations = config.token_location
+
+    # add the functions in the order specified in JWT_TOKEN_LOCATION
+    for location in locations:
+        if location == 'cookies' and config.jwt_in_cookies:
```
Comment (vimalloc) [inline flask_jwt_extended/view_decorators.py]: I think we can get rid of the `config.jwt_in_cookies` (et al) from these if statements, as they are doing the same check that we are doing here.

**Comment by vimalloc** [comment]
I think the way you have it where if there is a valid token higher up in the token location list supersedes an invalid token later in token location list makes the most sense :+1:

File: flask_jwt_extended/view_decorators.py (hunk 1 above)
Comment (stephendwolff) [reply flask_jwt_extended/view_decorators.py]: Good point!

**Comment by vimalloc** [comment]
Looks great! Thanks for contributing! I'll get a new version released momentarily.

**Pull request merged**